oak/server/oak_node.cc

```diff
@@ -261,13 +261,21 @@ std::unique_ptr<OakNode> OakNode::Create(const std::string& module) {
   // outlast this thread.
   LOG(INFO) << "Executing module oak_main on new thread";
   std::thread t([&node] { node->RunModule(); });
-  // TODO: join() instead when we have node termination
-  t.detach();
+  node->main_ = std::move(t);
   LOG(INFO) << "Started module execution thread";
 
   return node;
 }
 
+OakNode::~OakNode() {
+  if (main_.joinable()) {
+    // Cannot destroy the OakNode instance until the main thread terminates.
+    LOG(INFO) << "Await main thread termination before destroying OakNode instance";
```

Comment:
Hmm, there's still some sort of race here that crashes the unit test half the time.

Reply:
Found it &ndash; a lambda was capturing a reference to an ephemeral `std::unique_ptr<Node>`.  (Rust definitely prevents some categories of foot-gun.)